certbot-apache/tests/configurator_test.py

```diff
@@ -1819,25 +1813,35 @@ def test_current_version(self):
         self.config._openssl_version = '1.0.2a'
         self.assertTrue('old' in self.config.pick_apache_config())
 
-    def test_openssl_version_errors(self):
+    def test_openssl_version_warns(self):
         self.config._openssl_version = '1.0.2a'
         self.assertEqual(self.config.openssl_version(), '1.0.2a')
 
         self.config._openssl_version = None
         self.config.parser.modules['ssl_module'] = None
-        self.assertEqual(self.config.openssl_version(), None)
+        with mock.patch("certbot_apache._internal.configurator.logger.warning") as mock_log:
+            self.assertEqual(self.config.openssl_version(), None)
+            self.assertTrue("Could not find ssl_module" in mock_log.call_args[0][0])
 
         self.config.parser.modules['ssl_module'] = "/fake/path"
-        self.assertRaises(errors.PluginError, self.config.openssl_version)
+        with mock.patch("certbot_apache._internal.configurator.logger.warning") as mock_log:
+            # Check that correct logger.warning was printed
+            self.assertEqual(self.config.openssl_version(), None)
+            self.assertTrue("Unable to read" in mock_log.call_args[0][0])
 
-        contents_missing_openssl = "these contents won't match the regex"
-        with mock.patch("%s.open" % six.moves.builtins.__name__,
-            mock.mock_open(read_data=contents_missing_openssl)) as mock_file:
+        contents_missing_openssl = b"these contents won't match the regex"
+        with mock.patch("certbot_apache._internal.configurator."
+            "ApacheConfigurator._open_module_file") as mock_omf:
+            mock_omf.return_value = contents_missing_openssl
             with mock.patch("certbot_apache._internal.configurator.logger.warning") as mock_log:
                 # Check that correct logger.warning was printed
                 self.assertEqual(self.config.openssl_version(), None)
-                self.assertTrue("Could not find" in mock_log.call_args[0][0])
+                self.assertTrue("Could not find OpenSSL" in mock_log.call_args[0][0])
 
+    def test_open_module_file(self):
+        mock_open = mock.mock_open(read_data="testing 12 3")
+        with mock.patch("six.moves.builtins.open", mock_open):
+            self.assertEqual (self.config._open_module_file("/nonsense/"), "testing 12 3")
```

Comment:
Today I learned that Python lets you put a space between the name of a function and its parentheses when calling it.

Reply:
I knew this, but that's not why I put a space there, I'll fix it 😂